Demo/HelloWorld/Client/src/Program.cs

```diff
@@ -0,0 +1,212 @@
+                                  tasks,
+                                  resultsPerTask) =>
+                           {
+                             await Run(e,
+                                       p,
+                                       s,
+                                       tasks,
+                                       resultsPerTask);
+                           },
+                           endpoint,
+                           partition,
+                           input,
+                           nTasks,
+                           nResultsPerTask);
+
+    // Parse the command line parameters and call the function that represents the application
+    return await rootCommand.InvokeAsync(args);
+  }
+
+  internal static async Task Run(string endpoint,
+                                 string partition,
+                                 string input,
+                                 int    nTasks,
+                                 int    nResultsPerTask)
+  {
+    // Create gRPC channel to connect with ArmoniK control plane
+    var channel = GrpcChannelFactory.CreateChannel(new GrpcClient
+                                                   {
+                                                     Endpoint = endpoint,
+                                                   });
+
+    // Create client for task submission
+    var submitterClient = new Submitter.SubmitterClient(channel);
+
+    // Request for session creation with default task options and allowed partitions for the session
+    var createSessionReply = submitterClient.CreateSession(new CreateSessionRequest
+                                                           {
+                                                             DefaultTaskOption = new TaskOptions
+                                                                                 {
+                                                                                   MaxDuration = Duration.FromTimeSpan(TimeSpan.FromHours(1)),
+                                                                                   MaxRetries  = 2,
+                                                                                   Priority    = 1,
+                                                                                   PartitionId = partition,
+                                                                                 },
+                                                             PartitionIds =
+                                                             {
+                                                               partition,
+                                                             },
+                                                           });
+
+    // Initialize list of result ids. It will be useful retrieve data created by tasks
+    var resultIds = new List<string>(nTasks * nResultsPerTask);
+
+    var createTaskReply = await submitterClient.CreateTasksAsync(createSessionReply.SessionId,
```

Comment:
Not related to PR, but when I see this, it clearly  highlight the need to split payload and task submission.

Reply:
I agree